controller/src/beerocks/master/tasks/rdkb/rdkb_wlan_task.cpp

```diff
@@ -377,16 +377,18 @@ void rdkb_wlan_task::handle_event(int event_type, void *obj)
     case STEERING_REMOVE_SOCKET: {
         if (obj) {
             auto event_obj = (listener_general_register_unregister_event *)obj;
-            TASK_LOG(DEBUG) << "STEERING_REMOVE_SOCKET event was received";
+            if (is_bml_rdkb_wlan_listener_socket(event_obj->sd)) {
+                TASK_LOG(DEBUG) << "STEERING_REMOVE_SOCKET event was received";
 
-            if (!set_bml_rdkb_wlan_events_update_enable(event_obj->sd, false)) {
-                TASK_LOG(ERROR) << "fail in set_bml_rdkb_wlan_events_update_enable";
-            }
+                if (!set_bml_rdkb_wlan_events_update_enable(event_obj->sd, false)) {
+                    TASK_LOG(ERROR) << "fail in set_bml_rdkb_wlan_events_update_enable";
+                }
 
-            remove_bml_rdkb_wlan_socket(event_obj->sd);
+                remove_bml_rdkb_wlan_socket(event_obj->sd);
```

Comment:
Actually, I don't think this is necessary because it is already done in `remove_bml_rdkb_wlan_socket()`.
So I would recommend removing the function you added and remove also "set_bml_rdkb_wlan_events_update_enable()" since there is no point to set the 'events_update' flag if you remove the socket afterward on "remove_bml_rdkb_wlan_socket".
The only thing needs to be done here is to add return value (bool) to `remove_bml_rdkb_wlan_socket()` that indicates if the socket has been removed. If it has been removed, print the message to the log.

Reply:
This is a good point  , but then all steering APIs needs to be retested and this is not the purpose of this PR.
What i think needs to be done is to refactor RDKB task or to consider removing it completely.